dwds/lib/src/debugging/inspector.dart

```diff
@@ -465,30 +451,82 @@ function($argsString) {
     return _serverPathToScriptRef[uri];
   }
 
+  /// All the scripts in the isolate.
   Future<ScriptList> getScripts(String isolateId) async {
-    var scripts = await scriptRefs(isolateId);
+    var scripts = await _getScripts(isolateId);
     return ScriptList()..scripts = scripts;
   }
 
-  /// Returns all scripts in the isolate.
-  Future<List<ScriptRef>> scriptRefs(String isolateId) async {
+  Future<List<ScriptRef>> _getScripts(String isolateId) async {
     checkIsolate(isolateId);
-    var scripts = <ScriptRef>[];
-    for (var lib in isolate.libraries) {
-      // We can't provide the source for `dart:` imports so ignore for now.
-      if (lib.id.startsWith('dart:')) continue;
-      for (var script in (await _getLibrary(isolateId, lib.id)).scripts) {
-        scripts.add(ScriptRef(uri: script.uri, id: script.id));
+    // Guard against a possible race condition loading the scripts.
+    if (_scriptRefsCompleter != null) {
+      await _scriptRefsCompleter.future;
+    } else {
+      _scriptRefsCompleter = Completer();
+      try {
+        await _populateScriptCaches();
+        _scriptRefsCompleter.complete();
+      } catch (e) {
+        _scriptRefsCompleter.completeError(e);
+        rethrow;
+      }
+    }
+    return _scriptRefs.values.toList();
+  }
+
+  /// Request and cache <ScriptRef>s for all the scripts in the application.
+  ///
+  /// This populates [_scriptRefs], [_scriptIdToLibraryId] and
+  /// [_serverPathToScriptRef]. It is a one-time operation, because if we do a
+  /// reload the inspector will get re-created.
+  Future<void> _populateScriptCaches() async {
+    var libraryUris = isolate.libraries.map((lib) => lib.uri).toList();
```

Comment:
If we are on a recent enough SDK...

```suggestion
    var libraryUris = [ for(var lib in isolate.libraries) lib.uri ];
```